docs/docs.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+go run docs/book/src/update-docs/generate_docs.go
+
+cd docs/book/src/component-config-tutorial/testdata/project
+
+make build
```

Comment:
nits: what about merging the two lines:

```shell
(cd docs/book/src/component-config-tutorial/testdata/project; make build)
```

(Assuming in the future we may go to multiple tutorial directories and run the command..)

Reply:
We do not need the make build to do this one. 
The build will be made in the ci already before we call make generate
Locally the contributor should work with make generate so, IMO we should keep the same behaviour.

Reply:
Do we have the step to execute codegen for those tutorials in the CI?

I failed to find the place. Do you have the reference?

Reply:
We call in the CI make check-testdata, see: https://github.com/kubernetes-sigs/kubebuilder/blob/master/.github/workflows/testdata.yml#L27

Then, see that this target calls ./test/testdata/check.sh: https://github.com/kubernetes-sigs/kubebuilder/blob/master/Makefile#L120-L122

See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/check.sh#L17-L39

So, my thought, in the long run, is we have a makefile target that calls that run the go implementation to generate docs/ Also, add this one to make generate. And then, ensure that we are testing/checking for all PR changes if the docs were also generated/is updated accordingly. 

Is that make sense? Such as we do in SDK to verify the samples as well.